spaceros/Earthfile

```diff
@@ -113,31 +119,38 @@ spaceros-artifacts:
   END
 
   IF [ $SPACEROS_REPO_URL = "https://github.com/space-ros/space-ros.git" ] 
-    # run wget because main repo's host is known + it is public 
+    # run wget instead of `GIT CLONE`, because main repo's host is known + it is public 
     RUN wget https://raw.githubusercontent.com/space-ros/space-ros/${_GIT_COMMIT_HASH}/ros2.repos
     RUN wget https://raw.githubusercontent.com/space-ros/space-ros/${_GIT_COMMIT_HASH}/excluded-pkgs.txt
   ELSE
-    GIT CLONE --branch ${_GIT_COMMIT_HASH} ${SPACEROS_REPO_URL} .
+    GIT CLONE --branch ${SPACEROS_GIT_REF} ${SPACEROS_REPO_URL} .
   END
-  SAVE ARTIFACT ros2.repos
+
+  RUN mkdir src
```

Comment:
Lets assume that i built spaceros image and then pushed changes to process_sarif, and now I want to retrigger the build.
If we had this piece of code running in `+sources`, it would reuse cache and not include changes, because `ros2.repos` file did not change. `+spaceros-artifacts` does not use cache, so we always produce accurate `exact.repos` file. 

Check my replies below to understand "full picture" :)

Reply:
Your question is very valid and sparked idea for some middle-ground solution, but this did not work either :(.
Let me explain how it works right now (after changes in this PR).

**Scenario 1: New changes are available**:
1. `+spaceros-artifacts` gets triggered. It is never cached due to `--no-cache` statement. It clones `ros2.repos` and produces `exact.repos` with pinned versions
2. `+sources` task gets triggered. It sees that `exact.repos` has changed, so it proceeds and clones src files, and saves them locally.
3. `+workspace` is triggered. Since it inherits from `+sources` and `+sources` was executed, it also gets executed; it copies `src` directory from local to docker and builds all the stuff.
4. Rest of tasks get executed

**Scenario 2: No new changes are available**:
1. `+spaceros-artifacts` gets triggered. It is never cached due to `--no-cache` statement. It clones `ros2.repos` and produces `exact.repos` with pinned versions
2. `+sources` task gets triggered. It sees that `exact.repos` **was not** changed, so it skips 
3. `+workspace` task gets triggered. Since it inherits from `+sources` and `+sources` was cached, it is also cached
4. All other tasks are cached


What I've tried after your comment is to modify `+workspace` task to inherit from `+setup` instead of `+sources` , however then `src` directory gets re-created everytime with new timestamps, and earthly does not cache it, which results in execution of all downstream tasks.

Reply:
I know this setup is pretty convoluted, but I could not find easier way to satisfy both utilizing cache and pulling latest changes